h/templates/client/group_list.html

```diff
@@ -2,29 +2,60 @@
   <span class="dropdown-toggle"
         data-toggle="dropdown"
         role="button"
-        ng-switch on="groups.focused().public">
+        ng-switch on="groups.focused().public"
+        ng-click="toggleShareLink(undefined)">
     <i class="h-icon-public" ng-switch-when="true"></i>
     <i class="h-icon-group" ng-switch-default></i>
     {{groups.focused().name}}
     <i class="h-icon-arrow-drop-down"></i>
   </span>
   <ul class="dropdown-menu pull-right" role="menu">
-    <li ng-repeat="group in groups.all()"
-        ng-class="group.id == groups.focused().id? 'selected' : ''">
-      <a class="group-name"
-         href=""
-         ng-click="groups.focus(group.id)"
-         ng-switch on="group.public">
-        <i class="h-icon-public" ng-switch-when="true"></i>
-        <i class="h-icon-group" ng-switch-default></i>
-        {{group.name}}
-      </a>
-      <a ng-href="{{group.url}}" ng-if="group.url"
-         target="_blank" class="h-icon-link" title="Share this group"></a>
+    <li ng-repeat="group in sortedGroups()">
+      <div ng-class="{'group-item': true, selected: group.id == groups.focused().id}"
+           ng-click="groups.focus(group.id)">
+        <!-- the group icon !-->
+        <div class="group-icon-container" ng-switch on="group.public">
+          <i class="h-icon-public" ng-switch-when="true"></i>
+          <i class="h-icon-group" ng-switch-default></i>
+        </div>
+        <!-- the group name and share link !-->
+        <div class="group-details">
+          <div class="group-name-container">
+            <a class="group-name-link"
+               href=""
+               title="{{ group.public ? 'Show public annotations' : 'Show and create annotations in ' + group.name }}">
+               {{group.name}}
+            </a>
+          </div>
+          <div class="share-link-container" ng-click="stopClickPropagation($event)" ng-if="!group.public">
```

Comment:
Not sure the indirection of `stopClickPropagation` adds anything here. `ng-click="$event.stopPropagation()"` would achieve the same thing and is equally clear, I think.